### ISSUE TITLE
how to support remote-debugger like electron?

### ISSUE THREAD

**Issue by qiangf811** [tauri-apps/wry#786]
I have a plan to refactor an old application. It is a code editor application that must integrate devtools and support remote debugging.

The Chromium command line provided by electron previously supported remote debugging, but currently no related functions and APIs have been found in Tauri.

**Comment by wusyong** [comment]
Wry has [with_devtools](https://docs.rs/wry/latest/wry/webview/struct.WebViewBuilder.html#method.with_devtools). Is this what you looking for?

**Comment by qiangf811** [comment]
> Wry has [with_devtools](https://docs.rs/wry/latest/wry/webview/struct.WebViewBuilder.html#method.with_devtools). Is this what you looking for?

no,it just open the current webview's devtools，i want to get  a webSocketDebuggerUrl

**Comment by amrbashir** [comment]
you can use https://docs.rs/wry/latest/x86_64-pc-windows-msvc/wry/webview/trait.WebViewBuilderExtWindows.html#tymethod.with_additionl_browser_args and pass `--remote-debugging-port=999` but only available on Windows, other platforms don't have a similar API.

**Comment by qiangf811** [comment]
> you can use https://docs.rs/wry/latest/x86_64-pc-windows-msvc/wry/webview/trait.WebViewBuilderExtWindows.html#tymethod.with_additionl_browser_args and pass `--remote-debugging-port=999` but only available on Windows, other platforms don't have a similar API.

thanks!

**Comment by zozobalogh0817** [comment]
Hi, @amrbashir how can I use this method in my TauriApplication ? In the `main.rs` I did not managed to get it work, can you please help me out with this ?

**Comment by amrbashir** [comment]
@zozobalogh0817 this is still not published on tauri side, it is planned to be released in the upcoming 1.3 version. See https://github.com/tauri-apps/tauri/pull/5815

**Comment by zozobalogh0817** [comment]
@amrbashir Thanks for the quick reply, then I will take an eye on the releases 😄 and wait for the 1.3 excitedly
 
Till then, Happy engineering 🏗️ 🥳

**Comment by lanphan** [comment]
Hello @qiangf811 ,

Is it possible to move with_additional_browser_args from WebViewBuilderExtWindows to WebViewBuilder, so that we can run it on all platforms (including mobile)?

**Comment by FabianLars** [comment]
The only thing you'd get from that is that you don't have to use target specific cfg flags (the `#[cfg(target_os = "windows")])` stuff) as the function would not be actually operational on the other platforms since the respecitve webviews don't have a similar setting.
tldr: No

**Comment by raffaeler** [comment]
> you can use https://docs.rs/wry/latest/x86_64-pc-windows-msvc/wry/webview/trait.WebViewBuilderExtWindows.html#tymethod.with_additionl_browser_args and pass `--remote-debugging-port=999` but only available on Windows, other platforms don't have a similar API.

This is not true (anymore?), I could open the port on Ubuntu using this env variable.
It looks like the http API is not available, but you can even use the websocket protocol with the Chromium devtools on Tauri and works like a charm.
```
WEBKIT_INSPECTOR_HTTP_SERVER=127.0.0.1:9222
```

**Comment by FabianLars** [comment]
The comment was about the `additional_browser_args` setting, that's only available on Windows. tbh though, i didn't expect this to work (well) on linux as well, so that's nice to hear and probably something we should document.

**Comment by raffaeler** [comment]
Thanks @FabianLars, the comment misleaded me when it said:
> other platforms don't have a similar API

Anyway, I commented here to help the community and in the hope this gets considered to provide a better integration with the testing frameworks.

BTW I am looking to enable automated testing using Playwright on both Windows and Linux, but I didn't find a straight path.
Any pointer or hint would be very welcome.

**Comment by FabianLars** [comment]
>  Any pointer or hint would be very welcome.

All i can do is wish you good luck 😅

**Comment by raffaeler** [comment]
> All i can do is wish you good luck 😅

ouch, that's nor promising well.

Anyway:
- I read (and hope) the WebKit and Chromium websocket protocol is the same
- WebKit apparently does not provide the http API

If that is right, I "only" have to find a way to connect to an existing instance (aka open port) and the rest should just work.
